src/RequestProcessor.cs

```diff
@@ -117,6 +124,28 @@ internal StreamingMessage ProcessWorkerInitRequest(StreamingMessage request)
                 RemoteSessionNamedPipeServer.CreateCustomNamedPipeServer(pipeName);
             }
 
+            try
+            {
```

Comment:
I believe that you moved the installation of function app dependencies to `ProcessWorkerInitRequest`. However, the Functions Host has a 60 minute timeout for the worker response, so this will cause a timeout exception. Could you please ping the Team channel on why we need this change? 

Alternately, if the function app does not have any dependencies installed, we can start an async task in `ProcessWorkerInitRequest` to install the dependencies and send the response back to the Function Host.  Then, on the first function invocation, we can check if the task completed executing, if this is the case, we can continue with function invocation. Otherwise, wait until the task completes. What are your thoughts on this @michaelpeng36 @AnatoliB @davidmrdavid ?

Reply:
To clarify, the dependency download actually hasn't moved. Before and after my changes, the download happens on the first function load request. I have just moved some of the pre-initialization to the function load request, so that certain properties (like the functions app directory) are available during the metadata request which happens before function load requests. 

The worker already does kick off a task to install dependencies on the first function load request, it does not wait until all dependencies are installed to return.

Reply:
Thank you @andystaples for the clarification. 

Yes, in the first function invocation, we use the first PowerShell instance to download the function app dependencies and wait until the installation completes before invoking the function. The wait takes place here https://github.com/Azure/azure-functions-powershell-worker/blob/187ad3369a930f9f0acda258fa58618bf9059a27/src/DependencyManagement/DependencyManager.cs#L164

Reply:
That method waits for the dependency install task to complete, but that task is kicked off on the first function load request, using the method StartDependencyInstallationIfNeeded on line 230 of RequestProcessor.cs. Anyway, I think we're good here